packages/twenty-front/src/modules/object-record/record-field/meta-types/display/components/ActorFieldDisplay.tsx

```diff
@@ -5,15 +5,19 @@ import { isNonEmptyString } from '@sniptt/guards';
 
 export const ActorFieldDisplay = () => {
   const { fieldValue } = useActorFieldDisplay();
+  const isSoftDeletedWorkspaceMember =
+    fieldValue.workspaceMemberId && !fieldValue.workspaceMember;
 
-  const name = !fieldValue.workspaceMemberId
+  const name = isSoftDeletedWorkspaceMember
```

Comment:
Could be written more simply I think :)
```
  const name = fieldValue.workspaceMemberId && fieldValue.workspaceMember
    ? [
        fieldValue.workspaceMember.name.firstName,
        fieldValue.workspaceMember.name.lastName,
      ].filter(isNonEmptyString).join(' ')
    : fieldValue.name;
```

Reply:
Unless I'm mistaken we could factorize the name computation as follows:
```ts
  const getActorDisplayName = () => {
    if (!isDefined(fieldValue)) {
      throw new Error("FieldValue is undefined, should never occur");
    }
  
    if (isDefined(workspaceMember)) {
      const { firstName, lastName } = workspaceMember.name;
      return [firstName, lastName].filter(isNonEmptyString).join(" ");
    }
  
    return fieldValue.name;
  };
```
ps: Also would be checking fieldValue definition

Reply:
it because the properties CreateBy are of type ActorMetadata. I am not sure where in the codebase exactly, but when we softDelete a workspaceMemeber, the related ActorMetadata of the other tables are hard-deleted.

Reply:
>  but when we softDelete a workspaceMemeber, the related ActorMetadata of the other tables are hard-deleted. That's the other way to solve this ticket as I mentioned in the desc Another way we could also get into

I don't get why we would delete the fact that it comes from gmail but not the workspaceMemberId? It seems like there's something wrong in what we're doing no?